youtube_dl/extractor/jamendo.py

```diff
@@ -0,0 +1,117 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+from collections import namedtuple
+
+from ..compat import compat_urlparse
+from .common import InfoExtractor
+
+FormatData = namedtuple('FormatData', [
+    'format_id', 'sub_domain', 'ext', 'quality'])
+
+
+class JamendoIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?jamendo\.com/track/(?P<id>[0-9]+)/(?P<display_id>[\w-]+)'
+    _TEST = {
+        'url': 'https://www.jamendo.com/track/196219/stories-from-emona-i',
+        'md5': '6e9e82ed6db98678f171c25a8ed09ffd',
+        'info_dict': {
+            'id': '196219',
+            'display_id': 'stories-from-emona-i',
+            'ext': 'flac',
+            'title': 'Stories from Emona I',
+            'thumbnail': 're:^https?://.*\.jpg'
+        }
+    }
+
+    def _real_extract(self, url):
+        url_data = self._VALID_URL_RE.match(url)
+        track_id = url_data.group('id')
+        display_id = url_data.group('display_id')
+        webpage = self._download_webpage(url, display_id)
+
+        thumbnail = self._html_search_meta(
+            'image', webpage, 'thumbnail', fatal=False)
+        title = self._html_search_meta('name', webpage, 'title')
+
+        url_template = 'https://%s.jamendo.com/?trackid=%s&format=%s&from=app-97dab294'
+        format_data = [
+            FormatData(
+                format_id='mp31', sub_domain='mp3l', ext='mp3', quality=0),
+            FormatData(
+                format_id='mp32', sub_domain='mp3d', ext='mp3', quality=1),
+            FormatData(
+                format_id='ogg1', sub_domain='ogg', ext='ogg', quality=2),
+            FormatData(
+                format_id='flac', sub_domain='flac', ext='flac', quality=3),
+        ]
+        formats = [
+            {
+                'format_id': fd.format_id,
+                'url': url_template % (fd.sub_domain, track_id, fd.format_id),
+                'ext': fd.ext,
+                'quality': fd.quality
+            }
+            for fd in format_data
+        ]
+        self._check_formats(formats, video_id=display_id)
```

Comment:
Nope, but I'm not really sure if all formats are available in all formats, though.